### PR TITLE
[docs] Add log of mismatches

### PR DESCRIPTION
**Pull request by tonkosi**

### CODE REVIEW

File: docs/mismatches.md [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+    > lldb_eval::EvaluateExpression(ts1, "this == (TestStruct*)this", ...);
+    // OK (true)
+
+    > lldb_eval::EvaluateExpression(ts2, "this == (TestStruct*)this", ...);
+    // comparison of distinct pointer types ('TestStruct *' and 'TestStruct *')
+    ```
+
+    *   TODO: Investigate the exact reason for this bug. This could be related
+        to equality operator of `lldb::SBType` not working as expected.
+    *   Encountered: 2021 Apr 14 (reported by fuzzer while supporting evaluation
```
Comment (werat) [inline docs/mismatches.md:254]: Let's have consistent ordering of items. Right now "Bugs in lldb-eval" seems to be ordered from newest to oldest, but "Possible bugs in LLDB" is the other way around. I prefer from oldest to newest (i.e. oldest at the top, adding new items to the bottom)

**Review by werat — COMMENTED** [review]
Awesome, thanks for writing this up!

I have a few suggestions described below. I'd also name this file just "bugs.md", "mismatches" sounds too elaborate :)

File: docs/mismatches.md (hunk 1 above)
Comment (tonkosi) [reply docs/mismatches.md]: Reordered from oldest to newest.

**Review by tonkosi — COMMENTED** [review]
Sounds good, "bugs.md" it is :)

**Review by werat — APPROVED** [review]
Awesome, thanks!

**Pull request merged**